pkg/operator/controller/controller_dns_configmap.go

```diff
@@ -52,6 +52,7 @@ func desiredDNSConfigMap(dns *operatorv1.DNS, clusterDomain string, daemonsetRef
 	name := DNSConfigMapName(dns)
 	cm.Namespace = name.Namespace
 	cm.Name = name.Name
+	cm.SetOwnerReferences([]metav1.OwnerReference{dnsOwnerRef(dns)})
```

Comment:
I thought we couldn't do owner references between cluster-scoped and namespace-scoped resources; is it valid to do so but only a problem for garbage collection?

In any case, I think this owner reference gets overridden by `cm.SetOwnerReferences([]metav1.OwnerReference{daemonsetRef})` below.  Assuming the dns owner reference is valid, I think we want both owner references: dns owner reference so that `EnqueueRequestForOwner` enqueues the dns rather than the configmap, and a daemonset owner reference so that garbage collection deletes the configmap when the daemonset is deleted.  Also, we should set `Controller` to false in the `daemonsetRef`.

Reply:
>I thought we couldn't do owner references between cluster-scoped and namespace-scoped resources; is it valid to do so but only a problem for garbage collection?

Cluster scoped owners are okay. https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.13/#ownerreference-v1-meta

>In any case, I think this owner reference gets overridden by cm.SetOwnerReferences([]metav1.OwnerReference{daemonsetRef}) below. 

Good catch

>Assuming the dns owner reference is valid, I think we want both owner references: dns owner reference so that EnqueueRequestForOwner enqueues the dns rather than the configmap, and a daemonset owner reference so that garbage collection deletes the configmap when the daemonset is deleted.

The Corefile's existence doesn't depend upon the existence of the daemonset. In fact, the daemonset container depends on the ConfigMap.

>Also, we should set Controller to false in the daemonsetRef.

Based on the API docs I'm not sure I totally understand what the controller flag even means here... help?

Reply:
Fixed the overwriting but didn't change the controller bit.

Reply:
> Based on the API docs I'm not sure I totally understand what the controller flag even means here... help?

See the godoc for `EnqueueRequestForOwner`: "IsController if set will only look at the first OwnerReference with Controller: true."  I infer from that that leaving `Controller` to true on the daemonset owner reference could cause the watch to enqueue the daemonset instead of the dns.

Reply:
Interesting, thanks